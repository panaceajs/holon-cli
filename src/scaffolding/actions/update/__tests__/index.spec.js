const { resolve } = require('path');
const snapshot = require('../../../../utils/task-snapshot');
const task = require('../');

describe('actions update', () => {
  it('should create actions', async () => {
    await expect(
      snapshot(task, {
        cwd: resolve(__dirname, '__fixtures__', 'non-existing'),
        actionTypes: { anAction: 'anAction', anotherAction: 'anotherAction' }
      })
    ).resolves.toMatchSnapshot();
  });
  it('should complete actions', async () => {
    await expect(
      snapshot(task, {
        cwd: resolve(__dirname, '__fixtures__', 'existing'),
        actionTypes: { anAction: 'anAction', anotherAction: 'anotherAction' }
      })
    ).resolves.toMatchSnapshot();
  });
});
