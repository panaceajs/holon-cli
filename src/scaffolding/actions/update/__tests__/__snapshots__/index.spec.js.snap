// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions update should complete actions 1`] = `
"
--------------------------------------------------------------------------------
// ./actions/index.js
// Update
--------------------------------------------------------------------------------

import { createAction } from 'redux-actions';
import { AN_ACTION } from '../action-types';

export const anAction = createAction(AN_ACTION);


      ↓ ↓ ↓ ↓ ↓ ↓



import { createAction } from 'redux-actions';
import { AN_ACTION, ANOTHER_ACTION } from '../action-types';
export const anAction = createAction(AN_ACTION);
export const anotherAction = createAction(ANOTHER_ACTION);




"
`;

exports[`actions update should create actions 1`] = `
"
--------------------------------------------------------------------------------
// ./actions/index.js
// Update
--------------------------------------------------------------------------------

import { createAction } from 'redux-actions';
import { IGNORED_ACTION } from '../action-types';

export const ignoredAction = createAction(IGNORED_ACTION);


      ↓ ↓ ↓ ↓ ↓ ↓



import { createAction } from 'redux-actions';
import { IGNORED_ACTION, AN_ACTION, ANOTHER_ACTION } from '../action-types';
export const ignoredAction = createAction(IGNORED_ACTION);
export const anAction = createAction(AN_ACTION);
export const anotherAction = createAction(ANOTHER_ACTION);




"
`;
